### PR TITLE
Add timezone to CronWorkflowSpec

### PR DESCRIPTION
**Pull request by bobh66**
This PR adds support for the timezone attribute to the CronWorkflow object.

Please let me know if additional changes are needed.

### CODE REVIEW

File: src/hera/cron_workflow.py
Comment (flaviuvadan) [inline src/hera/cron_workflow.py:72]: Minor suggestion: 
```suggestion
        self.cron_spec = V1alpha1CronWorkflowSpec(schedule=self.schedule, timezone=self.timezone, workflow_spec=self.spec)
```

**Review by flaviuvadan — COMMENTED** [review]
Any chance you had the opportunity to functionally test this against an Argo cluster? 🙂  Also, need to:
- increment `VERSION`
- add a `CHANGELOG` entry

**Comment by flaviuvadan** [comment]
I've been thinking about the way we have validators in `Resource` and `Task` and I am wondering whether we should implement one for a `Workflow`/`CronWorkflow` as well. One simple validation is checking that the specified `timezone` field is valid on a `CronWorkflow`. This can be done by introducing `pytz`. I took a look through that package and doing a `pip show pytz` lists no additional dependencies (balance between implementing vs. bringing in a library). Perhaps on a different PR/issue, what are your thoughts on adding `pytz` so we can check whether `timezone in pytz.all_timezones`? Would that offer users an additional level of safety?

File: src/hera/cron_workflow.py
Comment (bobh66) [reply src/hera/cron_workflow.py:72]: I'll update the PR

**Comment by bobh66** [comment]
I was able to test these changes against an Argo server and verified that the timezone attribute shows up in the CronWorkflow object.  I didn't validate the timezone _functionality_ on the server (yet) but that would be a bug against the server if it's not working.

I'm doing the timezone validation in my application API (using a pydantic validator with pytz) but we could put it here too if you like.  I agree it's a balancing act and it's not always clear where the validation should happen.

I'll update the PR with the requested changes.

Thanks!

File: CHANGELOG.md
Comment (flaviuvadan) [inline CHANGELOG.md:36]: What do you think about including a small note about `pytz` so we know when it was introduced? 

```suggestion
- add support for the timezone attribute of CronWorkflow and validate the specified timezone
- introduce `pytz` dependency for timezone validation
```

File: Pipfile
Comment (flaviuvadan) [inline Pipfile:8]: There's no need to include it in the `Pipfile` as it is automatically brought in, and locked, through the `setup.py` specification 🙂  
```suggestion
```

File: tests/test_cron_workflow.py
Comment (flaviuvadan) [inline tests/test_cron_workflow.py:151]: Perhaps include a check for the error message to ensure we get a failing test in case it changes? 
```suggestion
def test_cwf_invalid_timezone_set(cws):
    with pytest.raises(ValueError) as e:
        cw = CronWorkflow('cw', "* * * * *", timezone="foobar", service=cws)
    assert str(e) == 'foobar is not a valid timezone'
```

File: src/hera/cron_workflow.py
Comment (flaviuvadan) [inline src/hera/cron_workflow.py:7]: I suggest we run a `make format` prior to merging so we get consistent formatting across the introduced changes! 🙂

**Comment by flaviuvadan** [comment]
Amazing stuff, thanks again for contributing! Left some minor suggestions that mostly have to do with cleanup and a test. We'll be ready to merge right after those are addressed 🚀

**Review by flaviuvadan — APPROVED** [review]

**Comment by flaviuvadan** [comment]
For the `Library stubs not installed for "pytz"` build error, we can add: 
```
[mypy-pytz.*]
ignore_missing_imports = True
```
to the `mypy.ini` file.

**Comment by flaviuvadan** [comment]
@bobh66 circling back to this 🙂  any chance we can get this shipped soon? After we adjust the `mypy` config, along with the `VERSION` and `CHANGELOG`!

**Comment by bobh66** [comment]
@flaviuvadan the tests fail when pytz is in setup.cfg but not in Pipfile.

Should I run pipenv install pytz locally and add Pipfile and Pipfile.lock into the PR?

**Comment by flaviuvadan** [comment]
> @flaviuvadan the tests fail when pytz is in setup.cfg but not in Pipfile.
> 
> Should I run pipenv install pytz locally and add Pipfile and Pipfile.lock into the PR?

Right! The build uses a virtual environment that has the packages listed in `Pipfile.lock` so we need to update the lockfile. The easiest way to do that is:
```shell
> deactivate // leave the current venv, if you are told you already have a shell, use `exit`
> pipenv --rm
> rm Pipfile.lock
> pipenv install --dev --pre
```
This will relock the project and bring in the specified dependencies from `setup.py` 🙂  Happy to explain more if needed! I think this is the last step we need.

**Comment by bobh66** [comment]
Thanks @flaviuvadan  - do I include Pipefile in the PR or just Pipfile.lock?

**Comment by flaviuvadan** [comment]
> Thanks @flaviuvadan - do I include Pipefile in the PR or just Pipfile.lock?

Just the `Pipfile.lock` changes!

**Review by flaviuvadan — APPROVED** [review]

**Comment by flaviuvadan** [comment]
Thank you for adding this @bobh66! 🚀  I am super thankful for all your patience in going back and forth on the design, review, and discussion!

**Pull request merged**
